File: lib/core/models/prescription.dart
// Prescription model — represents a doctor-issued prescription
class Prescription {
  final String id;
  final String doctorName;
  final String date;
  final List<String> medicines;
  final String notes;

  const Prescription({
    required this.id,
    required this.doctorName,
    required this.date,
    required this.medicines,
    required this.notes,
  });

  factory Prescription.fromJson(Map<String, dynamic> json) {
    final rawMedicines = (json['medicines'] as List<dynamic>?) ??
        (json['medications'] as List<dynamic>?);

    final List<String> list = rawMedicines?.map((e) {
          if (e is Map<String, dynamic>) {
            final name = e['name'] ?? '';
            final dosage = e['dosage'] ?? '';
            final freq = e['frequency'] ?? '';
            final instr = e['instructions'] ?? '';
            final parts = [
              name,
              if (dosage.isNotEmpty) dosage,
              if (freq.isNotEmpty) '($freq)',
              if (instr.isNotEmpty) '• $instr',
            ];
            return parts.join(' ');
          }
          return e.toString();
        }).toList() ??
        const [];

    return Prescription(
      id: json['id'] as String? ?? '',
      doctorName: json['doctorName'] as String? ?? '',
      date: json['date'] as String? ?? '',
      medicines: list,
      notes: json['notes'] as String? ?? json['diagnosis'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'doctorName': doctorName,
      'date': date,
      'medicines': medicines,
      'notes': notes,
    };
  }
}

File: backend/src/models/Prescription.js
const mongoose = require('mongoose');

const medicationSchema = new mongoose.Schema(
    {
        name: { type: String, required: true },
        dosage: { type: String, required: true },
        frequency: { type: String, required: true },
        duration: { type: String, required: true },
        instructions: { type: String, default: '' },
    },
    { _id: false }
);

const prescriptionSchema = new mongoose.Schema(
    {
        patientId: {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'Patient',
            required: true,
            index: true,
        },
        doctorName: { type: String, required: true },
        doctorSpecialty: { type: String, default: 'General Medicine' },
        facilityName: { type: String, required: true },
        medications: { type: [medicationSchema], default: [] },
        diagnosis: { type: String, default: '' },
        notes: { type: String, default: '' },
        prescribedDate: { type: Date, default: Date.now },
        validUntil: { type: Date },
    },
    {
        timestamps: true,
        collection: 'prescriptions',
    }
);

module.exports = mongoose.model('Prescription', prescriptionSchema);
